wasm2c/wasm-rt-impl.c

```diff
@@ -172,14 +172,41 @@ static void os_signal_handler(int sig, siginfo_t* si, void* unused) {
   }
 }
 
-static void os_install_signal_handler(void) {
-  /* Use alt stack to handle SIGSEGV from stack overflow */
+static bool os_has_altstack_installed() {
+  /* check for altstack already in place */
+  stack_t ss;
+  if (sigaltstack(NULL, &ss) != 0) {
+    perror("sigaltstack failed");
+    abort();
+  }
+
+  return !(ss.ss_flags & SS_DISABLE);
+}
+
+/* Use alt stack to handle SIGSEGV from stack overflow */
+static void os_allocate_and_install_altstack(void) {
+  /* verify altstack not already allocated */
+#ifndef NDEBUG
+  if (g_alt_stack) {
+    fprintf(
+        stderr,
+        "wasm-rt error: tried to re-allocate thread-local alternate stack\n");
+    abort();
+  }
```

Comment:
I'm not sure what the convention is within this file, but would it make sense to write these 6 lines as `assert(!g_alt_stack && "wasm-rt error: tried to re-allocate thread-local alternate stack")`?

Reply:
Sure, done.